### PR TITLE
[JENKINS-70658] Update maven goal detection for Maven 3.9.0 

### PR DESCRIPTION
**Pull request by Dneth04**
Maven 3.9.0 changed the logging format of goals and maven plugins, see [JENKINS-70658](https://issues.jenkins.io/browse/JENKINS-70658). The PR updates all parsers that work on Maven logs.

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira

### CODE REVIEW

File: src/main/java/edu/hm/hafner/analysis/parser/MavenConsoleParser.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ protected boolean isLineInteresting(final String line) {
 
     private boolean isValidGoal() {
         return !(goal.contains("maven-compiler-plugin")
-                || goal.contains("maven-javadoc-plugin")); // will be captured by another parser already
+                || goal.contains("maven-javadoc-plugin") || goal.contains("compiler")); // will be captured by another parser already
     }
```
[the remark this answers is not in the document]
Comment (Dneth04) [reply src/main/java/edu/hm/hafner/analysis/parser/MavenConsoleParser.java:89]: Okay will try that !

[the remark this answers is not in the document]
Comment (Dneth04) [reply src/main/java/edu/hm/hafner/analysis/parser/MavenConsoleParser.java:89]: Will this code work to test whether each line is parsed by only one parser ?

public class MyMavenRunner {

    public static void main(final String[] args) throws IOException {
        // Defining the parsers to use
        MavenConsoleParser mavenConsoleParser = new MavenConsoleParser();
        JavacParser javacParser = new JavacParser();
        JavaDocParser javaDocParser = new JavaDocParser();

        // Runs the Maven build and capture its logs
        String mavenLogs = runMavenBuild();

        // Splits the logs into individual lines
        String[] logLines = mavenLogs.split("\n");

        // Iterates over each line in the logs and check which parser is able to extract information from it
        for (String logLine : logLines) {
            boolean parsedByMavenConsoleParser = mavenConsoleParser.isLineInteresting(logLine);
            boolean parsedByJavacParser = javacParser.isLineInteresting(logLine);
            boolean parsedByJavaDocParser = javaDocParser.isLineInteresting(logLine);

            // To Count the number of parsers that were able to parse the line
            int num = (parsedByMavenConsoleParser ? 1 : 0) + (parsedByJavacParser ? 1 : 0) + (parsedByJavaDocParser ? 1 : 0);

            // Checks that only one parser was able to parse the line
            if (num== 0) {
                System.err.println("Warning: line \"" + logLine + "\" was not parsed by any parser");
            } else if (num > 1) {
                System.err.println("Warning: line \"" + logLine + "\" was parsed by multiple parsers");
            } 
	    else{
		    System.out.println("The logLine is parsed by only one parser");} 
        }
    }

File: SUPPORTED-FORMATS.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<!--- DO NOT EDIT -- Generated at 2023-02-16T14:25:13.065416 - Run the `main` method of `ParserRegistry` to regenerate after changing parsers -- DO NOT EDIT --->
+<!--- DO NOT EDIT -- Generated at 2023-02-25T20:47:50.473344898 - Run the `main` method of `ParserRegistry` to regenerate after changing parsers -- DO NOT EDIT --->
```
Comment (uhafner) [inline SUPPORTED-FORMATS.md:1]: Remove this unrelated change from the PR.

**Review by uhafner — CHANGES_REQUESTED** [review]
Normally you should *first* create a failing test that exposes the problem. Then you start developing a fix.

Please add some tests that show that your fix actually works.

You also should update all classes that derive from `AbstractMavenLogParser`.

File: src/main/java/edu/hm/hafner/analysis/parser/MavenConsoleParser.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dneth04) [reply src/main/java/edu/hm/hafner/analysis/parser/MavenConsoleParser.java:89]: I am really sorry since I am stuck with few of my assignment deadlines from my university, I am not able to focus into this .Will look into the issue again on Friday.

**Comment by uhafner** [comment]
Superseded by #917
